src/main/java/org/commcare/suite/model/MenuLoader.java

```diff
@@ -145,12 +140,48 @@ private boolean menuIsRelevant(SessionWrapperInterface sessionWrapper, Menu m) t
         return true;
     }
 
+    public boolean menuAssertionsPass(SessionWrapperInterface sessionWrapper, Menu m) throws XPathSyntaxException {
+        AssertionSet assertions = m.getAssertions();
+        Vector<String> assertionXPathStrings = assertions.getAssertionsXPaths();
+        if (!assertionXPathStrings.isEmpty()) {
+
+            for (int i = 0; i < assertionXPathStrings.size(); i++) {
+                XPathExpression assertionXPath = XPathParseTool.parseXPath(assertionXPathStrings.get(i));
+                EvaluationContext traceableContext = accumulateInstances(sessionWrapper, m, assertionXPath);
+
+                Text text = assertions.evalAssertionAtIndex(i, assertionXPath, traceableContext);
+
+                InstrumentationUtils.printAndClearTraces(traceReporter, "menu load expand");
```

Comment:
probably  "menu assertions" is a better description here than "menu load expand".

Reply:
updated here https://github.com/dimagi/commcare-core/pull/1267/commits/4b427aba9e8f5f4f7e695dcaa20a47559a0c71f2